### DISCUSSION TITLE
Aggiorno a Python 3.9.0 ma HA legge Python 3.8.6

### DISCUSSION THREAD

**Discussion by manny199** [indomus/forum#1035] [HUB: Home Assistant]
Ciao a tutti, questo è il terzo Raspberry che aggiorno ma in questo caso, dopo avere eseguito la procedura per il passaggio da Python 3.8.6 a 3.9.0, alla fine mi compare la norifica "Support for the running Python version 3.8.6 is deprecated and will be removed in Home Assistant 2022.1. Please upgrade Python to 3.9.0 or higher".
Effettivamente, guardando nella sezione informazioni, python risulta ancora nella versione 3.8.6.
Prima di fare di nuovo l'aggiornamento, onde evitare casini, chiedo cosa può essermi sfuggito?
in \home\pi la cartella python 3.9.0 é presente.
Se eseguo il comando python3 -V, la risposta è Python 3.9.0.
Mi chiedo perchè HA invece vede 3.8.6

Grazie

**Comment by Luxor73** [comment]
Docker

**Comment by snakuzzo** [comment]
Ha ricreato il venv con la 3.8.x 
Ad ogni modo ti consiglio di lasciar perdere venv e passare a Docker (come consigliato da @Luxor73).
Con Docker non dovrai mai pià occuparti di tutto questo

**Comment by Scialla** [comment]
Tu hai installato la 3.9 ma non hai ricreato il venv con la 3.9, probabilmente, seguendo la guida, hai ricreato il venv con la 3.8.6. I comandi della guida vanno adattati con la nuova versione di Python

**Reply by manny199** [reply]
Infatti, maledetto copia incolla ...
In merito al passaggio su docker, in termini di prestazioni, cambia qualcosa?

**Reply by snakuzzo** [reply]
no...vedrai che le performance saranno uguali, se non migliori
